### PR TITLE
Cleaned up documentation.

### PR DESCRIPTION
**Pull request by daemonmaker**
@lamblin In this case the parent class doesn't necessarily have the parameters so I copied the documentation from other sources.

### CODE REVIEW

File: pylearn2/models/mlp.py [hunk 1]
```diff
@@ -1516,20 +1516,27 @@ class SoftmaxPool(Layer):
...
+    layer_name : str
+        The name of the layer. All layers in an MLP must have a unique name.
     pool_size : WRITEME
-    irange : WRITEME
-    sparse_init : WRITEME
+    irange : float
+        If specified, initialized each weight randomly in U(-irange, irange).
+    sparse_init : int
+        If specified, initial sparse_init number of weights for each unit from
+        N(0,1).
```
Comment (lamblin) [inline pylearn2/models/mlp.py]: Could you specify that `irange` and `sparse_init` are optional in the type?

File: pylearn2/models/mlp.py [hunk 2]
```diff
@@ -1516,20 +1516,27 @@ class SoftmaxPool(Layer):
...
-    W_lr_scale : WRITEME
-    b_lr_scale : WRITEME
+    W_lr_scale : float, optional
+        Multiply the learning rate on the weights by this constant.
+    b_lr_scale : float, optional
+        Multiply the learning rate on the biases by this constant.
     mask_weights : WRITEME
-    max_col_norm : WRITEME
+    max_col_norm : float
+        Maximum norm for a column of the weight matrix.
```
Comment (lamblin) [inline pylearn2/models/mlp.py]: Same here for `max_col_norm`.

File: pylearn2/models/mlp.py [hunk 3]
```diff
@@ -1886,8 +1897,10 @@ class Linear(Layer):
     mask_weights : ndarray, optional
         If provided, the weights will be multiplied by this mask after each
         learning update.
-    max_row_norm : WRITEME
-    max_col_norm : WRITEME
+    max_row_norm : float
+        Maximum norm for a row of the weight matrix.
+    max_col_norm : float
+        Maximum norm for a column of the weight matrix.
```
Comment (lamblin) [inline pylearn2/models/mlp.py]: Same for that class.

**Comment by lamblin** [comment]
Apart from that, it's good, thanks!

**Comment by daemonmaker** [comment]
@lamblin I have incorporated your feedback.

**Comment by lamblin** [comment]
Great, thanks!

**Pull request merged**